src/main/java/io/cdap/e2e/pages/actions/CdfPluginPropertiesActions.java

```diff
@@ -852,4 +869,33 @@ public static void verifyRadioButtonPluginPropertySelectedValue(String pluginPro
   public static void clickBrowseButton() {
     ElementHelper.clickOnElement(CdfPluginPropertiesLocators.browseButton);
   }
+
+  /**
+   * Enter KeyValue Pairs For Property
+   *
+   * @param pluginProperty    @data-cy attribute value of Plugin Property.
+   *                          If pluginProperty is present in {@link ConstantsUtil#DEFAULT_DATACY_ATTRIBUTES_FILE}
+   *                          then its data-cy is fetched from it
+   *                          else pluginProperty is used as it is.
+   * @param jsonKeyValuePairs Actual json KeyValue Pairs string is fetched
+   *                          from {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} with jsonKeyValuePairs as key
+   */
+  public static void enterKeyValuePairsForProperty(String pluginProperty, String jsonKeyValuePairs) {
+    String pluginPropertyDataCyAttribute = PluginPropertyUtils.getPluginPropertyDataCyAttribute(pluginProperty);
+    if (pluginPropertyDataCyAttribute == null) {
+      pluginPropertyDataCyAttribute = pluginProperty;
+    }
+    Map<String, String> properties =
+      JsonUtils.convertKeyValueJsonArrayToMap(PluginPropertyUtils.pluginProp(jsonKeyValuePairs));
+    int index = 0;
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      ElementHelper.sendKeys(CdfPluginPropertiesLocators.locatePropertyKey(
+        pluginPropertyDataCyAttribute, index), entry.getKey());
+      ElementHelper.sendKeys(CdfPluginPropertiesLocators.locatePropertyValue(
+        pluginPropertyDataCyAttribute, index), entry.getValue());
+      ElementHelper.clickOnElement(CdfPluginPropertiesLocators.locatePropertyAddRowButton(
```

Comment:
I think this will end up adding an extra row than required?

Reply:
Updated logic to add row.